cmd/aad-cli/cli/user_test.go

```diff
@@ -0,0 +1,275 @@
+
+			if slices.Contains(args, "--name") {
+				username := args[slices.Index(args, "--name")+1]
+				user, err := cache.GetUserByName(context.Background(), username)
+				require.NoError(t, err, "Setup: failed to get user from cache")
+				// Timestamps get serialized as RFC3339 which includes timezone
+				// information.
+				// We replace this with the unix timestamp to make
+				// the comparison easier.
+				got = testutils.TimestampToUnix(t, got, user.LastOnlineAuth)
+			}
+
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "expected output to match golden file")
+		})
+	}
+}
+
+func TestUserSetAttribute(t *testing.T) {
+	tests := map[string]struct {
+		args string
+
+		badPerms bool
+		wantErr  bool
+	}{
+		"set gecos": {args: "user --name futureuser@domain.com gecos newvalue"},
+		"set home":  {args: "user --name futureuser@domain.com home newvalue"},
+		"set shell": {args: "user --name futureuser@domain.com shell newvalue"},
+
+		// error cases
+		"set bad_attribute":    {args: "user --name futureuser@domain.com bad_attribute newvalue", wantErr: true},
+		"set nonexistent user": {args: "user --name nouser@domain.com gecos newvalue", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			args := strings.Split(tc.args, " ")
+			username := args[slices.Index(args, "--name")+1]
+
+			cacheDir := t.TempDir()
+			cacheDB := "db_with_old_users"
+			testutils.PrepareDBsForTests(t, cacheDir, cacheDB)
+			cache := testutils.NewCacheForTests(t, cacheDir)
+			c := cli.New(cli.WithCache(cache))
+
+			_, err := testutils.RunApp(t, c, args...)
+			if tc.wantErr {
+				require.Error(t, err, "expected command to return an error")
+				return
+			}
+			require.NoError(t, err, "expected command to succeed")
+
+			user, err := cache.GetUserByName(context.Background(), username)
+			require.NoError(t, err, "Setup: failed to get user from cache")
+			got, err := user.IniString()
+			require.NoError(t, err, "Setup: failed to get user representation as ini")
+			got = testutils.TimestampToUnix(t, got, user.LastOnlineAuth)
+
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "expected output to match golden file")
+		})
+	}
+}
+
+func TestUserMoveHomeDirectory(t *testing.T) {
+	tests := map[string]struct {
+		prevHomeDir string
+		newHomeDir  string
+
+		wantErr bool
+	}{
+		"move home directory": {prevHomeDir: "oldhome", newHomeDir: "newhome"},
+
+		// Error cases
+		"fail if previous directory is absent": {prevHomeDir: "absent", newHomeDir: "newhome", wantErr: true},
+		"fail if previous directory is a file": {prevHomeDir: "oldhomefile", newHomeDir: "newhome", wantErr: true},
+		"fail if new directory already exists": {prevHomeDir: "oldhome", newHomeDir: "existingnewhome", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			cacheDir := t.TempDir()
```

Comment:
Mind renaming this tmpDir and having the cache in a subdirectory. I find it weird to create the home in cache :)